### PR TITLE
Add ChatGPT Code Review Capabilities

### PR DESCRIPTION
**Pull request by trustlreis**
This pull request introduces an automated code review process using ChatGPT. The workflow is configured to trigger on pull request events (`opened` and `synchronize`) and will utilize the `cirolini/genai-code-review` action to perform the code review. This addition aims to improve the code review process by leveraging AI to provide initial feedback on pull requests.

### Changes Made:
- Added a GitHub Actions workflow file named `.github/workflows/chatgpt-code-review.yml` to set up ChatGPT code review capabilities.

### Motivation and Context:
Automating the code review process using ChatGPT will streamline the review process, providing quicker feedback on pull requests. This will help maintain code quality and reduce the manual effort required from maintainers.

### How It Works:
1. **Trigger**: The workflow is triggered on pull request events (`opened` and `synchronize`).
2. **Job**: The job `code_review_job` runs on `ubuntu-latest`.
3. **Step**: The step `GenAI Code Review` uses the `cirolini/genai-code-review` action.
   - The action uses the OpenAI API to perform a code review.
   - It utilizes the `OPENAI_TOKEN` and `GITHUB_TOKEN` from the repository secrets.
   - The OpenAI model used is `gpt-4o` with a temperature setting of `0.3` and a maximum token limit of `2048`.

### Checklist:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have documented my changes in the `.github/workflows/chatgpt-code-review.yml` file.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have checked my code and corrected any misspellings.

### CODE REVIEW

**Review by thalesminussi — APPROVED** [review]

**Review by thalesminussi — APPROVED** [review]

**Comment by trustlreis** [comment]
AI Code Review:

Based on the provided pull request's code, I can provide the following analysis regarding optimization:

1. `chatgpt-review.yml`:
   - The file mainly consists of defining a workflow for ChatGPT code review using GitHub Actions.
   - There are some commented-out steps related to GenAI Code Review, and the current steps involve checking out the repository and performing a code review using the `manjunathsb100/chatgpt-code-review-action` action.
   - No noticeable optimization is needed for this file as it seems appropriately structured for its purpose.

2. `sonar_cloud.yml`:
   - This file defines a workflow for building and analyzing the project using SonarCloud on GitHub Actions.
   - The main optimization opportunity here lies in the `run` section where the Maven command is executed. This command could potentially be improved by breaking it down into multiple lines for better readability and maintainability.
   - Instead of having the Maven command all in one line, splitting it into separate lines using the YAML multiline syntax can make it easier to read and manage.
   - The optimization change involves breaking the run command into separate lines for better clarity, as shown below:

Before:
```sh
run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=trustlreis_candidate-test-stub-maven
```

After:
```sh
run: |
    mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar \
        -Dsonar.projectKey=trustlreis_candidate-test-stub-maven
```

Overall, the optimization suggestion is to enhance the readability and manageability of the `sonar_cloud.yml` file by breaking down the Maven command into multiple lines for better clarity.

File: .github/workflows/chatgpt-review.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: AI Code Reviewer
+        uses: hohyon-ryu/ai-codereviewer@code_review
+        with:
+          GITHUB_TOKEN: ${{ secrets.CUSTOM_GITHUB_TOKEN }} # The GITHUB_TOKEN is there by default so you just need to keep it like it is and not necessarily need to add it as secret as it will throw an error. [More Details](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret)
+          OPENAI_API_KEY: ${{ secrets.OPENAI_TOKEN }}
```
Comment (trustlreis) [inline .github/workflows/chatgpt-review.yml]: The change from `OPENAI_API_KEY` to `OPENAI_TOKEN` might cause confusion. Please ensure that the correct key is being used and that the repository secrets are updated accordingly.

File: .github/workflows/chatgpt-review.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    name: ChatGPT Code Review
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: AI Code Reviewer
+        uses: hohyon-ryu/ai-codereviewer@code_review
+        with:
+          GITHUB_TOKEN: ${{ secrets.CUSTOM_GITHUB_TOKEN }} # The GITHUB_TOKEN is there by default so you just need to keep it like it is and not necessarily need to add it as secret as it will throw an error. [More Details](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret)
```
Comment (trustlreis) [inline .github/workflows/chatgpt-review.yml]: The comment here is misleading. The `GITHUB_TOKEN` is automatically provided by GitHub Actions and does not need to be added as a secret. The current comment suggests adding it as a secret, which is incorrect. Consider rephrasing to clarify that the `GITHUB_TOKEN` is automatically available and should not be manually added as a secret.

File: .github/workflows/chatgpt-review.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          OPENAI_API_KEY: ${{ secrets.OPENAI_TOKEN }}
+          OPENAI_API_MODEL: "gpt-4o" # Optional: defaults to "gpt-4"
```
Comment (trustlreis) [inline .github/workflows/chatgpt-review.yml]: The model name `gpt-4o` seems unusual. Ensure that this is the correct model name supported by the OpenAI API. If `gpt-4o` is a typo, it should be corrected to `gpt-4`.

**Comment by trustlreis** [comment]
AI Code Review:

Upon analyzing the pull request's code, I have identified potential areas that may require optimization:

1. In the `.github/workflows/chatgpt-review.yml` file:
    - The code includes commented-out steps that are not necessary for execution. It is recommended to remove unnecessary commented code to enhance code readability and maintainability.
    - There is an error in the step 'Checkout Repor' which appears to be a typo (should be 'Checkout Repo'). This typo should be corrected to ensure the step functions correctly.

2. In the `.github/workflows/sonar_cloud.yml` file:
    - The code includes an unnecessary extra space in the 'runs-on' attribute. While not a critical issue, it can be considered for improvement to maintain code consistency.
    - The `run` command is split into multi-line format using `|` for better readability, which is a good practice.

3. In the `SampleTest.java` file:
    - The test case `testAddition` contains an incorrect assertion message, stating that "2 + 2 should equal 5." This assertion is incorrect and should be updated to reflect the correct expected result.

In conclusion, while there are some minor areas for improvement in the provided code, they do not significantly impact performance or efficiency. The optimizations suggested are mainly focused on enhancing code readability and correctness.

**Comment by trustlreis** [comment]
AI Code Review:

Upon analyzing the provided code changes from the pull request, here is my assessment:

1. **chatgpt-review.yml**:
   - This file adds a workflow for ChatGPT code review.
   - It triggers on pull request events for opened and synchronized types.
   - It uses GitHub Actions for the workflow.
   - The steps include checking out the repository and running the `chatgpt-code-review-action` with specified inputs.
   - No optimization is required as the workflow seems to be properly defined and straightforward for ChatGPT code review.

2. **sonar_cloud.yml**:
   - This file adds a workflow for SonarCloud analysis.
   - It triggers on push and pull request events.
   - It runs on the `ubuntu-latest` environment.
   - The steps include checking out the repository and running the Sonar scanner using Maven.
   - The Maven command has been slightly formatted to be more readable.
   - No significant optimization is required for this workflow unless specific performance issues are encountered during execution.

3. **SampleTest.java**:
   - This file adds a JUnit test class for testing addition functionality.
   - It contains a simple JUnit test method `testAddition` that checks if 2 + 2 equals 4.
   - The test message in the `assertEquals` method seems incorrect as it says "2 + 2 should equal 5" instead of 4.
   - No optimization is required in terms of the test logic; however, the test message should be corrected for accuracy.

Overall, from an optimization perspective:
- The workflows seem well-structured and defined for their intended purposes.
- No specific performance bottlenecks or inefficiencies are evident in the provided code changes.
- The JUnit test seems fine logic-wise, but the test message could be optimized for correctness.

In conclusion, optimization is not necessary for the provided code changes, but minor improvements could be made for clarity and correctness in the test case's message.
